crypto/bls12381/key.go

```diff
@@ -0,0 +1,111 @@
+//go:build !bls12381 || !((linux && amd64) || (linux && arm64) || (darwin && amd64) || (darwin && arm64) || (windows && amd64))
+
+package bls12381
+
+import (
+	"github.com/cometbft/cometbft/crypto"
+)
+
+const (
+	// PrivKeySize defines the length of the PrivKey byte array.
+	PrivKeySize = 32
+	// PubKeySize defines the length of the PubKey byte array.
+	PubKeySize = 48
+	// SignatureLength defines the byte length of a BLS signature.
+	SignatureLength = 96
+	// KeyType is the string constant for the bls12_381 algorithm.
+	KeyType = "bls12_381"
+	// Enabled indicates if this curve is enabled.
+	Enabled = false
+)
+
+// -------------------------------------.
+const (
+	PubKeyName = "cometbft/PubKeyBLS12_381"
+)
+
+// ===============================================================================================
+// Private Key
+// ===============================================================================================
+
+// PrivKey is a wrapper around the Ethereum bls12_381 private key type. This
+// wrapper conforms to crypto.Pubkey to allow for the use of the Ethereum
+// bls12_381 private key type.
+
+// Compile-time type assertion.
+var _ crypto.PrivKey = &PrivKey{}
+
+// PrivKey represents a BLS private key noop when blst is not set as a build flag and cgo is disabled.
+type PrivKey []byte
+
+func NewPrivateKeyFromBytes([]byte) (PrivKey, error) {
+	panic("bls12_381 is disabled")
```

Comment:
Would it be better to return it as an error, rather than panicking? (same comment below)

Reply:
Sure